### DISCUSSION TITLE
Update the App Icon

### DISCUSSION THREAD

**Discussion by patmauro** [H-uru/Plasma#1391] [Ideas & Proposals]
Uru's application icon dates all the way back to 2000, being the exact same icon used for 2000's realMyst. That was 23 years and _seven_ Windows operating systems ago. I think a little refresh may be overdue.
![MOULa](https://user-images.githubusercontent.com/6642931/236309455-04417f6f-a01c-4fc7-9fa1-b502ab869b86.png)

Obviously this does not apply to the Gehn client, which uses a spiffy, custom GoW logo in place of the standard app icon (which even changes design to optimize based on size - nice!). But I think the mainline Huru client could use some similar love, and hopefully, this could be something that could be absorbed into MOULa itself in the future.

As far as what icon to use - I'd suggest adopting the app icon used for the short-lived Mac client, which sported a lovingly-made Relto book graphic:
![UruIconActualSize](https://user-images.githubusercontent.com/6642931/236311886-f655f096-240d-4ab4-ac0e-6a01e13d1eb6.png)

Possible downsides to this:
- While still an improvement, still not as modern as it should be - this Mac icon is _itself_ on the older side, so even though it is much bigger than the classic icon, it's still a max size of 128x128, and not 256x256 (or ideally even up to 512 or 1024) like it should be. However Cyan still uses variants of this graphic for other apps and may be able to provide an upscaled version.
- Sits a little low for an icon?

Comparison of how each of these hold up at "medium" vs "extra large" icon size in Windows (incl. Gehn client for reference):
![IconCompare](https://user-images.githubusercontent.com/6642931/236317271-d54c1316-8421-4fc2-9383-782f15736e7a.png)

For reference - I've attached a ZIP with the original .ICNS file ripped from the old Mac client and a rudimentary .ICO conversion. (I've been using this ICO for a few of the clients to help differentiate them and they work pretty well.) [UruIcons.zip](https://github.com/H-uru/Plasma/files/11401171/UruIcons.zip)

Thoughts on this?

**Comment by dpogue** [comment]
I had been thinking of emailing Cyan and asking if they would be willing to provide a 1024x1024 version of the macOS Uru icon. It matches the icon style used in a bunch of their newer Myst/Riven releases.

I've not reached out to anyone yet because I know they're in the final sprint for Firmament.

**Comment by Deledrius** [comment]
I've always been a fan of the Mac icon and would love to see a full res version adopted if Cyan is willing.

**Comment by TheodoreEvans** [comment]
This is very interesting information! I've really enjoyed reading your points, and I think you're right about a lot of them. You are amazing.  I should thank you for all the work you put into making this post [five nights at freddy's](https://fngames.io/)

**Comment by patmauro** [comment]
Using App store assets and some Photoshop, it is possible to extrapolate an accurate reproduction of the old OSX icon at 256, the full size currently used by Windows:
![UruIcon256 fw](https://github.com/H-uru/Plasma/assets/6642931/84202d57-7e52-4dd1-85cd-88b482b7bac9)
As you can see, this holds up rather well:
![image](https://github.com/H-uru/Plasma/assets/6642931/049fb058-eab3-4a84-af23-c213c9607dcd)

So this is a complete, accurate conversion of the old OSX icon that is ready to go for Windows. Really the only question we need to ask is: do we _like_ this old OSX icon as-is, and want to replicate it _directly,_ or do we want to make any changes while we're at it? For example, centering the icon so it floats a little less low - it really does look a bit odd. Here is Centered vs Original, with GoW/Gehn for comparison.
![image](https://github.com/H-uru/Plasma/assets/6642931/1e6457a7-e000-4019-b085-b05f6110f823)

I've attached both original and centered ICOs in a zip:
[UruIcons_256.zip](https://github.com/H-uru/Plasma/files/11618517/UruIcons_256.zip)

**Comment by patmauro** [comment]
So I think we have everything we need - what would be the process to move this forward? Would this be a new PR that replaces all instances of **dirt.ICO** contained in the Plasma repo? Or are there only specific instances of this we need to worry about?

At first I wanted to suggest a similar change on OU as well, but actually, I'm thinking this might serve as a nice shorthand for differentiating the legacy & H'uru clients for now. Plus, once the OU client eventually _does_ catch up with H'uru, it will make it that much more apparent to end users that the client has undergone a significant upgrade.

**Comment by dpogue** [comment]
This looks great for Windows client, but it still poses an issue for macOS where we're expected to provide a 1024⨉1024 icon (and newer macOS guidelines want it to be in a solid coloured box). I think without an original source asset (or a very high-res copy of one) we're going to run into issues long-term.

That maybe shouldn't stop us from updating the Windows icon though.

**Comment by patmauro** [comment]
> This looks great for Windows client, but it still poses an issue for macOS where we're expected to provide a 1024⨉1024 icon (and newer macOS guidelines want it to be in a solid coloured box). I think without an original source asset (or a very high-res copy of one) we're going to run into issues long-term.

I have my doubts about whether an original source asset even exists at that size or greater, to be honest. We can ask, but note how Cyan uses derivatives of this icon on iOS, but notably _not_ on macOS anymore. I suspect if we want this, it may be up to us to re-create something at this size on our own. I can actually take a crack at this - I had some success with an upscale test, and have some ideas for how we could adapt this to be "macOS guideline compliant" if we want to go down that road. I'll do some mockups and follow up. That being said -

> That maybe shouldn't stop us from updating the Windows icon though.

Yeah, as much as I agree with the desire to make sure we're more future-proof here, and can take this as a separate action item, an updated Windows icon is nonetheless ready to go here. So for Windows at least - what would we need to do to move forward? Again, would this entail replacing all instances of "dirt.ico" in the Plasma repo, or is there a better way?

**Comment by dpogue** [comment]
> Yeah, as much as I agree with the desire to make sure we're more future-proof here, and can take this as a separate action item, an updated Windows icon is nonetheless ready to go here. So for Windows at least - what would we need to do to move forward? Again, would this entail replacing all instances of "dirt.ico" in the Plasma repo, or is there a better way?

Yeah, I think the two important ones are for the main player-facing apps:
* DIRT.ico here for the main client: https://github.com/H-uru/Plasma/tree/master/Sources/Plasma/Apps/plClient/win32/res
* DIRT.ico here for the launcher/updater binary: https://github.com/H-uru/Plasma/tree/master/Sources/Plasma/Apps/plUruLauncher

**Comment by patmauro** [comment]
Have created https://github.com/H-uru/Plasma/pull/1411 covering the Windows icon specifically. Still to be done:
- Check with Cyan re: an original source asset
- Create an upscaled replica
- Create macOS & Linux -compliant icons
- Figure out how to compile with these icons in the other clients as needed (?)

**Comment by patmauro** [comment]
Took a pass at an upscaled replica, and while it's plenty big (the book measures 1803x1319 before drop shadow), I'm not really happy with the results so far - the book and stamp are mostly fine, but I can't seem to get the linking panel quite right. I'll attempt another pass at some point. Probably also worth checking with Cyan to see if they still have the source img for this, and how big it is.

However, I did want to take a second to share a proof-of-concept for a potential macOS-compliant icon, using the guidelines I could find and taking some inspiration from the way Cyan has branded their iPadOS apps. (I rendered the transparent edges here as dark gray for demonstration purposes - Apple currently recommends a ~100px transparent border around square icons.) Would love your input @dpogue @colincornaby - what do we think?
![macOsIcon_ProofOfConcept fw](https://github.com/H-uru/Plasma/assets/6642931/8967761c-1e00-413e-a282-80620269225c)

**Reply by Deledrius** [reply]
The upscaling itself definitely has severe issues, but as a proof-of-concept I think it's a good indicator of how this could work.  Hopefully we can get a high-res source from Cyan.  Otherwise, it might be worth recreating it instead.  Upscaling tools really still aren't up to the task without showing their flaws.

We'll definitely want this in some form that can be adjusted as Apple's guidelines change in the future to follow (or lead) whatever stylistic fads come around next for icons.

**Comment by patmauro** [comment]
Circling back to this per https://github.com/H-uru/Plasma/pull/1420. Have dropped a line to Cyan, specifically Eric A. Anderson, asking if they have the original full-size source asset as suggested by @dpogue, so we'll see what they come back with. In the event that I get anything back from them, I'll see about submitting this to moul-assets so we have the source image on file going forward.

Regarding the Mac icon: in a way, I almost have complicated feelings about updating the Mac icon to be compliant with Apple's current suggested best practices, rather than just updating the 2007 one so it supports higher resolutions (Apple's standards change so frequently; how long until the _current ones_ change and we have to do this all over again?) - that being said, assuming this is what we want to do, I'll reiterate that I think the style I proposed earlier would be the way to go. Not only does it incorporate the original icon, but it's visually consistent with Cyan's other App icons that used derivatives of this same book graphic - in other words, it _feels_ like what Cyan probably would have done were they still supporting this app directly.

Thinking ahead - what will be needed for **Linux?** I take it something closer to the Windows icon will be acceptable here rather than Apple's highly-specific requirements, but what format will this need to be prepared in, etc?

**Reply by dpogue** [reply]
For Linux a PNG is probably fine. There's not a tonne of consistency across icons on Linux due to different desktop environments and different eras of design guidelines, so using the high-res Windows icon is probably fine.

**Reply by dgelessus** [reply]
No strong opinions from me about whether the Mac icon needs to match the current (macOS 11+) icon design. Looking through my Applications folder, there are a number of things with icons that *don't* match the current design, but they're indeed mostly older applications that haven't been updated. There are some exceptions though (Skype, VLC, JetBrains IDEs, Pacifist).

I don't really understand the complaints about Apple changing the icon design often... isn't this the first major design change since the introduction of Mac OS X? Unless you count the transition to an overall "flat" look in OS X 10.10, but that's also 9 years ago already, and didn't matter as much to third-party icons I think.

Linux uses plain PNGs with a specific name/directory structure for multiple resolutions. You can mostly get away with just one high resolution and let the desktop environment scale it down - but I remember that (a few years ago at least) some parts of KDE expect specific resolutions to exist, or the icon won't be displayed at all.

**Reply by patmauro** [reply]
I heard back from Eric A. Anderson - he is checking internally if they have the asset and will let us know.

Re: Apple - This was more an observation than a complaint; all in all, I _do_ actually think updating it is the way to go here. Any anxiety I have here is probably just from musing about the difficulty in trying to future-proof an application if you can't predict what standards will change in the future and when - but really, such is the life of any developer. :)

**Comment by patmauro** [comment]
Hi everyone, got the official word from Cyan on this. Short version is **unfortunately, no, they are unable to provide a high-resolution source asset for this** - this was outsourced to a contractor they were working with at the time - so we're on our own as far as creating a higher-resolution replica of this graphic. He was able to send over a slightly higher resolution version of the "clean" Myst book used for the iOS apps, but this it was already the same resolution as one I was already able to track down on my own, so we're back to where we started.
![MystBookIcon](https://github.com/H-uru/Plasma/assets/6642931/7f514864-0971-42ff-9436-d3e53d5922fb)

With that mystery solved, and the Windows icon already updated, I think we might actually be able to close this thread - for the macOS and Linux icons, we can probably handle that in the relevant PRs for those.

**Comment by Kohtlyr** [comment]
Just made a larger version, not sure if this would suffice.

![relto-book-icon-lrg](https://github.com/H-uru/Plasma/assets/89598313/2219d636-7cd8-4586-a44e-f8140d418ca8)

**Reply by Deledrius** [reply]
Nice texture on this one!

**Comment by patmauro** [comment]
We should have everything we need here now. Will proceed with development in the relevant PRs.
